src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -404,6 +394,25 @@ internal Symbol BindNamespaceOrTypeOrAliasSymbol(ExpressionSyntax syntax, Diagno
             }
         }
 
+        private Symbol BindTupleExpression(TupleTypeSyntax syntax, DiagnosticBag diagnostics)
+        {
+            var count = syntax.Elements.Count;
+            var symbols = ArrayBuilder<TypeSymbol>.GetInstance(count);
+            var names = ArrayBuilder<string>.GetInstance(count);
+
+            for (int i = 0; i < count; i++)
+            {
+                var argumentSyntax = syntax.Elements[i];
```

Comment:
If Elements is a syntax list, it probably has very efficient enumerator implementation as a structure. You should check.